yt/fields/magnetic_field.py

```diff
@@ -194,11 +194,61 @@ def setup_magnetic_field_aliases(registry, ds_ftype, ds_fields, ftype="gas"):
     ...             setup_magnetic_field_aliases
     ...         setup_magnetic_field_aliases(self, "chombo", ["bx%s" % ax for ax in [1,2,3]])
     """
-    unit_system = registry.ds.unit_system
     ds_fields = [(ds_ftype, fd) for fd in ds_fields]
-    if ds_fields[0] not in registry:
+    convert, units = _setup_magnet_unit_conversion(registry, ds_fields[0])
+    def mag_field(fd):
+        def _mag_field(field, data):
+            return convert(data[fd])
+        return _mag_field
+    for ax, fd in zip(registry.ds.coordinates.axis_order, ds_fields):
+        registry.add_field((ftype,"magnetic_field_%s" % ax),
+                           sampling_type="cell", 
+                           function=mag_field(fd),
+                           units=units)
+
+def setup_particle_magnetic_field_aliases(registry, ds_ptype, ds_field):
```

Comment:
I think for particle datasets we always want the resulting particle type to be the same as the particle type to be aliased. So the last argument is removed as compared with `setup_magnetic_field_aliases`. I'd like to know others' opinions on this.

Reply:
Is there not a way we can make `setup_magnetic_field_aliases` do this job also by merging the functionality of these two functions into one and just give it appropriate arguments?

Reply:
Yes, I think it's doable.